### PR TITLE
Fix preview link

### PR DESCRIPTION
**Pull request by el-mapache**
* Updates default s3 bucket computation to handle staging env
* Adds preview link to individual site item on `/sites`
* Adds preview link to `View this Website` buttons in the `pagesHeader` component

### CODE REVIEW

File: api/controllers/PreviewController.js [hunk 1]
```diff
@@ -16,20 +16,19 @@ module.exports = {
...
-          if (redirect) return res.redirect(redirect);
-          res.set(headers);
-        }),
-        stream = object.createReadStream().on('error', function(error) {
-          var file = key.split('/').pop().indexOf('.') !== -1,
-              notFound = error.statusCode === 404;
-          if (!file && notFound) return res.redirect(req.path + '/');
-          res.send(error.statusCode, error.message);
-        }).pipe(res);
+      Bucket: sails.config.build.s3Bucket,
```
Comment (el-mapache) [inline api/controllers/PreviewController.js]: The indentation was bugging me here, so I changed it

File: api/models/Site.js [hunk 2]
```diff
@@ -5,7 +5,11 @@
 * @docs        :: http://sailsjs.org/#!documentation/models
 */
 
-var DEFAULT_BUCKET = 'federalist.18f.gov.s3-website-us-east-1.amazonaws.com';
+var env = process.env.NODE_ENV;
+
+var DEFAULT_BUCKET = env === 'staging' ?
+  'federalist-staging.18f.gov.s3-website-us-east-1.amazonaws.com' :
+  'federalist.18f.gov.s3-website-us-east-1.amazonaws.com';
```
Comment (brandocalrissian) [inline api/models/Site.js]: Do we want to make the bucket configuration come from the environment so we don't have this configuration/switch in code? If we always get the settings from the same environment variable, we don't need to know if we're dev/staging/production or whatever.

I'm fine keeping this for now as we had the production config here already; it's something to think about.

Comment (el-mapache) [inline api/models/Site.js]: Yeah that is a valid point.

Comment (brandocalrissian) [inline api/models/Site.js]: This is my only comment! Everything looks good!

I'm happy with merging this PR and making the "pull the bucket from the env" another (small) card/ticket/issue/task/whatever we're calling it.

Comment (el-mapache) [inline api/models/Site.js]: I think we should go ahead and merge this, we'll still need some kind of mechanism for ensuring the user can get redirected to one of these URLs when the app is running locally (like manually setting env variables) and I don't want to have more moving parts to keep track of at this point

Comment (brandocalrissian) [inline api/models/Site.js]: Sounds good. Merging.

**Pull request merged**